runtime/src/accounts_db.rs

```diff
@@ -1381,9 +1416,68 @@ impl AccountsDB {
             .and_then(|account_storage_entry| account_storage_entry.get_account(account_info))
     }
 
-    fn find_storage_candidate(&self, slot: Slot) -> Arc<AccountStorageEntry> {
+    fn try_recycle_and_insert_store(
+        &self,
+        slot: Slot,
+        min_size: u64,
+        max_size: u64,
+    ) -> Option<Arc<AccountStorageEntry>> {
+        let store = self.try_recycle_store(slot, min_size, max_size)?;
+        self.insert_store(slot, store.clone());
+        Some(store)
+    }
+
+    fn try_recycle_store(
+        &self,
+        slot: Slot,
+        min_size: u64,
+        max_size: u64,
```

Comment:
is there any reason to have a `max_size` if we're always passing `std::u64::MAX`?

Reply:
When shrinking, we don't pass in u64::MAX.